pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml

```diff
@@ -0,0 +1,673 @@
+                  name: avi-k8s-config
+                  key: nsSyncLabelValue
+            - name: DEFAULT_ING_CONTROLLER
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: defaultIngController
+            - name: SEG_NAME
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: serviceEngineGroupName
+            - name: BGP_PEER_LABELS
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: bgpPeerLabels
+            - name: NODE_NETWORK_LIST
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: nodeNetworkList
+            - name: VIP_NETWORK_LIST
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: vipNetworkList
+            - name: AKO_API_PORT
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: apiServerPort
+            - name: SERVICE_TYPE
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: serviceType
+            - name: ADVANCED_L4
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: advancedL4
+            - name: AUTO_L4_FQDN
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: autoFQDN
+            - name: LOG_FILE_PATH
+              value: ""
+            - name: LOG_FILE_NAME
+              value: ""
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          resources:
+            limits:
+              cpu: "250m"
+              memory: "300Mi"
+            requests:
+              cpu: "100m"
+              memory: "100Mi"
+          livenessProbe:
+            httpGet:
+              path: /api/status
+              port:  8080
+            initialDelaySeconds: 10
+            periodSeconds: 10
+#@ end
+
+#@ def ako_ingressclass():
+#@ if not data.values.AVI_DISABLE_INGRESS_CLASS:
+apiVersion: networking.k8s.io/v1
+kind: IngressClass
+metadata:
+  name: avi-lb
+  #@ if data.values.AVI_INGRESS_DEFAULT_INGRESS_CONTROLLER:
+  annotations:
+    ingressclass.kubernetes.io/is-default-class: "true"
+  #@ end
+spec:
+  controller: ako.vmware.com/avi-lb
+  parameters:
+    apiGroup: ako.vmware.com
+    kind: IngressParameters
+    name: external-lb
+#@ end
+#@ end
+
+#@ def hostrules_crd():
```

Comment:
Could you also try to explore do we really need all of those following CRDs here?